check_gha_pinning/__init__.py

```diff
@@ -106,12 +129,14 @@ def check_pinning(file: pathlib.Path) -> list[str]:
                     hash = tags.get(ref)
                     if not hash:
                         raise _RefNotFoundError(f"tag {ref} not found")
-                    tags = shas.get(hash) if hash else [ref]
+                    tags = shas.get(hash)
```

Comment:
If an action is pinned to commit `ref` who does not belong to a tag then `_get_action_tags(action)` will return `({}, {})`. `tags` would be `None` and the line below would fail. That's why I added the fallback to the `[ref]`.

Reply:
although now I'm thinking that the _RefNotFoundError would be raised anyways... Did we lose the ability to pin to non-tagged commits in https://github.com/lalten/check-gha-pinning/commit/905e704dbfd196c80fa641b278deab542e0b1dbc?

Reply:
Ooh. that could have been dumb. I'll look!

Reply:
Thinking about this:

If we're tagged to a SHA already, `_NotPinnedToCommitError` won't be thrown in `_check`:

```
    elif not re.match(_SHA1, line.split("@")[1]):
        raise _NotPinnedToCommitError()
```

Which means that the whole exception handler won't be called:
https://github.com/whi-tw/check-gha-pinning/blob/5c3c21a940ce3760f98a87ef0bd4ede757a4b87f/check_gha_pinning/__init__.py#L122-L137

If we've reached [this point](https://github.com/whi-tw/check-gha-pinning/blob/5c3c21a940ce3760f98a87ef0bd4ede757a4b87f/check_gha_pinning/__init__.py#L129), we've definitely got a non-sha-tagged action.

Unsure if that makes sense / if I'm understanding this right.

I've added an integration test with a sha-tagged commit to see what happens.

Reply:
I'm thinking it might make sense for a refactoring PR after this one to make the logic a little easier to follow :D